frontend/src/components/RequireAuth.tsx

```diff
@@ -0,0 +1,32 @@
+import { useGetCurrentUserAuthorizationQueryOverride } from '@hooks/useGetCurrentUserAuthorizationQueryOverride'
+import { getOIDCConfig } from 'auth/getOIDCConfig'
+import { useAuth } from 'react-oidc-context'
+import { Navigate } from 'react-router-dom'
+
+export function RequireAuth({ children, redirect = false, requireSuperUser = false }) {
+  const oidcConfig = getOIDCConfig()
+  const auth = useAuth()
+  const { data: user } = useGetCurrentUserAuthorizationQueryOverride(undefined, { skip: !auth?.isAuthenticated })
+
+  if (!oidcConfig.IS_OIDC_ENABLED) {
```

Comment:
suggestion: un peu de refacto des conditions, sinon je m'y perds un peu. Qu'en penses-tu ?
```suggestion
  if (!oidcConfig.IS_OIDC_ENABLED) {
    return children;
  }

  if (!auth.isAuthenticated) {
    return handleRedirect("/login", redirect);
  }


  if (requireSuperUser && !user?.isSuperUser) {
    return handleRedirect("/register", redirect);
  }

  return children;
}

const handleRedirect = (path, redirect) => {
  if (redirect) {
    return <Navigate replace to={path} />;
  }
  return null;
}
```